components/artist_page_cta/view.coffee

```diff
@@ -20,8 +26,28 @@ module.exports = class ArtistPageCTAView extends Backbone.View
     @$el.removeClass 'fullscreen'
     @render()
 
+  submit: (e) ->
+    return unless @validateForm()
+    return if @formIsSubmitting()
+
+    e.preventDefault()
+
+    @$('button').attr 'data-state', 'loading'
+
+    @user.set (data = @serializeForm())
+    @user.signup
+      success: @onRegisterSuccess
+      error: (model, response, options) =>
+        @reenableForm()
+        message = @errorMessage response
+        mediator.trigger 'auth:error', message
+
+  onRegisterSuccess: (model, response, options) =>
```

Comment:
Ideally successful registration (either Facebook or with email), as well as existing user login via modal can trigger this, which will swap out this content in the overlay and swap in/initialize the follow artist widgy